### DISCUSSION TITLE
Unable to create an initial migration, "Unrecognized keys: [databases];"

### DISCUSSION THREAD

**Discussion by TheSHEEEP** [simolus3/drift#3291] [Q&A] [answered]
I am trying to follow this example to create migrations for my database automatically:
https://drift.simonbinder.eu/migrations/

However, that guide is either incomplete or I am doing something wrong (or both).

This is my build.yaml, taken straight from the guide:
```
targets:
  $default:
    builders:
      drift_dev:
        options:
          databases:
            # Required: A name for the database and it's path
            OfflineDatabase: lib/database/offline_database.dart
```
The name of the database and the path to its .dart file is correct.

However, when running `dart run drift_dev make-migrations`, I get the following error:
```
Building package executable... 
Built drift_dev:drift_dev.
Unhandled exception:
CheckedFromJsonException
Could not create `DriftOptions`.
Unrecognized keys: [databases]; supported keys: [write_from_json_string_constructor, override_hash_and_equals_in_result_sets, skip_verification_code, use_data_class_name_for_companions, use_column_name_as_json_key_when_defined_in_moor_file, use_sql_column_name_as_json_key, generate_connect_constructor, generate_manager, sqlite_modules, sqlite, sql, data_class_to_companions, mutable_classes, raw_result_set_data, apply_converters_on_variables, generate_values_in_copy_with, named_parameters, named_parameters_always_required, scoped_dart_components, store_date_time_values_as_text, case_from_dart_to_sql, write_to_columns_mixins, assume_correct_reference, has_separate_analyzer, preamble, fatal_warnings]
#0      $checkedNew (package:json_annotation/src/checked_helpers.dart:66:5)
#1      $checkedCreate (package:json_annotation/src/checked_helpers.dart:30:10)
#2      _$DriftOptionsFromJson (package:drift_dev/src/generated/analysis/options.g.dart:9:50)
#3      new DriftOptions.fromJson (package:drift_dev/src/analysis/options.dart:205:46)
#4      readOptionsFromConfig.<anonymous closure> (package:drift_dev/src/utils/options_reader.dart:25:39)
#5      MappedIterator.moveNext (dart:_internal/iterable.dart:403:20)
#6      IterableExtension.firstOrNull (package:collection/src/iterable_extensions.dart:265:18)
#7      readOptionsFromConfig (package:drift_dev/src/utils/options_reader.dart:26:12)
#8      new DriftProject (package:drift_dev/src/cli/project.dart:19:19)
#9      DriftProject.readFromDir (package:drift_dev/src/cli/project.dart:40:12)
<asynchronous suspension>
#10     DriftDevCli.run (package:drift_dev/src/cli/cli.dart:65:15)
<asynchronous suspension>
#11     run (package:drift_dev/src/cli/cli.dart:19:12)
<asynchronous suspension>

```
I also tried using the schema instead, but then a very similar error appears with drift_dev not recognizing key `schema_dir`.  
Creating the schema itself from the .dart file worked fine.

Not really sure how to proceed here.
I could imagine that something is missing from my build.yaml file - I did not have such a file before as I never needed it - and that the guide assumes something else that would have to be set or done beforehand.

I could of course create manual migrations, but I'd rather avoid that if anyhow possible.

**Comment by simolus3** [comment, marked as answer]
Could you check that you're on `drift_dev` version 2.21.0? The `make-migrations` command was added in that version.

If you're on 2.20.0 or below, you can still follow [this](https://drift.simonbinder.eu/migrations/exports/) and [this](https://drift.simonbinder.eu/migrations/step_by_step/) to generate essentially the same thing, the `make-migrations` command essentially combines the two steps for you.

**Reply by TheSHEEEP** [reply]
Oh, I'm actually on 2.20.3 - just my luck, I guess. Missed the new version by one day 😅

After the upgrade to 2.21.0 the make-migrations command runs through just fine.
